src/main/java/com/google/sps/servlets/PreviousTerms.java

```diff
@@ -0,0 +1,114 @@
+package com.google.sps.servlets;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/prev-terms")
+public class PreviousTerms extends HttpServlet {
+  private DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<Entity> prevTermsData = getPreviousTerms(db, request);
+    String prevTermsDataJSON = makeJSON(prevTermsData);
+    response.setContentType("application/json;");
+    response.getWriter().println(prevTermsDataJSON);
+  }
+
+  private List<Entity> getPreviousTerms(DatastoreService db, HttpServletRequest request) {
+    String schoolName = request.getParameter("school-name");
+    String courseName = request.getParameter("course-name");
+    Long units = Long.parseLong(request.getParameter("num-units"));
+    Integer comparisonCount = Integer.parseInt(request.getParameter("comparison-count"));
```

Comment:
"comparison-count" implies that you're making comparisons but really it seems more accurate to say it's a limit for how many previous terms you return. I'd recommend renaming to "term-limit" or something like that